core/core.go

```diff
@@ -124,38 +135,39 @@ func New(config Config) (*App, error) {
 		}
 	}()
 
-	// Initialize order watcher (but don't start it yet).
-	orderWatcher, err := orderwatch.New(db, blockWatcher, ethClient, config.EthereumNetworkID, config.OrderExpirationBuffer)
+	// Initialize the order validator
+	orderValidator, err := zeroex.NewOrderValidator(ethClient, config.EthereumNetworkID, config.EthereumRPCMaxContentLength)
 	if err != nil {
 		return nil, err
 	}
 
-	// Initialize the ETH balance watcher (but don't start it yet).
-	ethWatcher, err := ethereum.NewETHWatcher(ethWatcherPollingInterval, ethClient, config.EthereumNetworkID)
+	// Initialize order watcher (but don't start it yet).
+	orderWatcher, err := orderwatch.New(db, blockWatcher, orderValidator, config.EthereumNetworkID, config.OrderExpirationBuffer)
 	if err != nil {
 		return nil, err
 	}
-	// TODO(albrow): Call Add for all existing makers/signers in the database.
 
-	// Initialize the order validator
-	orderValidator, err := zeroex.NewOrderValidator(ethClient, config.EthereumNetworkID)
+	// Initialize the ETH balance watcher (but don't start it yet).
+	ethWatcher, err := ethereum.NewETHWatcher(ethWatcherPollingInterval, ethClient, config.EthereumNetworkID)
```

Comment:
Annoying that the diff is messing up here. I literally just move the `OrderValidator` instantiation above the `OrderWatcher` instantiation since we now pass it into the `OrderWatcher` instead of re-instatiating it within it.

Reply:
🤷‍♂thanks for the heads up.